File: src/main/scala/com/github/mauricio/postgresql/messages/frontend/PreparedStatementMessage.scala
package com.github.mauricio.postgresql.messages.frontend

import com.github.mauricio.postgresql.column.ColumnEncoderDecoder

/**
 * User: Maurício Linhares
 * Date: 3/12/12
 * Time: 10:12 PM
 */

class PreparedStatementMessage( kind : Char, val query : String, val values : Seq[Any] ) extends FrontendMessage(kind) {

  val valueTypes : Seq[Int] = values.map {
    value =>
      if ( value == null ) {
        0
      } else {
        ColumnEncoderDecoder.kindFor(value.asInstanceOf[AnyRef].getClass)
      }

  }

}

File: postgresql-async/src/main/java/com/github/jasync/sql/db/postgresql/messages/frontend/PreparedStatementMessage.kt
package com.github.jasync.sql.db.postgresql.messages.frontend

import com.github.jasync.sql.db.column.ColumnEncoderRegistry
import java.util.UUID

open class PreparedStatementMessage(
    val statementId: UUID,
    override val kind: Int,
    val query: String,
    val values: List<Any?>,
    val encoderRegistry: ColumnEncoderRegistry
) : ClientMessage(kind) {
    val valueTypes: List<Int> = values.map { value ->
        encoderRegistry.kindOf(value)
    }
}
